contracts/Interfaces/ILockedFund.sol

```diff
@@ -98,19 +117,51 @@ contract ILockedFund {
 
 	/**
 	 * @notice Creates vesting contract (if it hasn't been created yet) for the calling user.
+	 * @return _vestingAddresses The New Vesting Contracts Created.
+	 * @dev Zero (0) is passed to denote all vesting for that user will be created.
+	 */
+	function createAllVesting() external returns (address[] memory _vestingAddresses);
+
+	/**
+	 * @notice Creates vesting contract (if it hasn't been created yet) for the calling user.
+	 * @param _vestingData The vesting details like cliff & duration in short form.
 	 * @return _vestingAddress The New Vesting Contract Created.
 	 */
-	function createVesting() external returns (address _vestingAddress);
+	function createVesting(bytes32 _vestingData) external returns (address[] memory _vestingAddress);
 
 	/**
 	 * @notice Stakes tokens for a user who already have a vesting created.
 	 * @dev The user should already have a vesting created, else this function will throw error.
+	 * @dev Zero (0) is passed to denote all vesting for that user will be created.
+	 * @dev This is not recommended function if there are many stakes to be created due to gas limit.
 	 */
-	function stakeTokens() external;
+	function stakeAllTokens() external;
+
+	/**
+	 * @notice Stakes tokens for a user who already have a vesting created.
+	 * @param _vestingData The vesting details like cliff & duration in short form.
+	 * @dev The user should already have a vesting created, else this function will throw error.
+	 */
+	function stakeTokens(bytes32 _vestingData) external;
 
 	/**
 	 * @notice Withdraws unlocked tokens and Stakes Locked tokens for a user who already have a vesting created.
 	 * @param _receiverAddress If specified, the unlocked balance will go to this address, else to msg.sender.
 	 */
 	function withdrawAndStakeTokens(address _receiverAddress) external;
+
+	/**
+	 * @notice Function to read the cliff, duration and Type of a Vesting.
+	 * @param _vestingData The address whose cliff and duration has to be found.
+	 * @return The cliff of the user vesting/lock.
+	 * @return The duration of the user vesting/lock.
+	 */
+	function getCliffDurationAndType(bytes32 _vestingData)
+		external
+		view
+		returns (
+			uint256,
+			uint256,
+			uint256
```

Comment:
add return vars names for clarity and 3rd return param description to func natspec

Reply:
Solved in e5c213e